File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution
{
    public:
        bool isPalindrome(int x)
        {
            if (x < 0) return false;
            long long num = x;
            long long digit = 0;

            while (num != 0)
            {
                int sum = num % 10;
                num = num / 10;
                digit = digit * 10 + sum;
            }

            return x == digit;
        }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int num = Math.abs(x);
        int rev = 0;
        
        while(num != 0) {
            int dig = num % 10;
            num = num / 10;
            
            rev = (10 * rev) + dig;
        }
        
        return rev == x;
    }
}